ui/src/components/Toolbox.tsx

```diff
@@ -0,0 +1,150 @@
+import { MouseEventHandler, ReactNode, useState } from 'react';
+import Return from '@icons/Return.svg?react';
+import Rubber from '@icons/Rubber.svg?react';
+import Pencil from '@icons/Pencil.svg?react';
+import Palette from '@icons/Palette.svg?react';
+import Download from '@icons/Download.svg?react';
+import classNames from 'classnames';
+import { useOutsideClick } from '@lib/clickOutside.tsx';
+import { drawingMode } from '../consts/drawingMode.ts';
+import { DrawMode } from '../types/types.ts';
+import { t } from 'i18next';
+
+interface ToolboxProps {
+  canDownload: boolean;
+  onErase: () => void;
+  onRevert: () => void;
+  onDraw: () => void;
+  onColorChange: (newColor: string) => void;
+  onDownload: () => void;
+}
+
+interface ToolboxItemProps {
+  icon: ReactNode;
+  text: string;
+  active: boolean;
+  onClick?: MouseEventHandler<HTMLDivElement>;
+}
+
+interface ToolboxColorPickerProps {
+  onColorChange: (newColor: string) => void;
+  handleClose: () => void;
+}
+
+interface ToolboxColorPickerElementProps {
+  color: string;
+  colorClass: string;
+  onColorChange: (newColor: string) => void;
+  handleClose: () => void;
+}
+
+const ToolboxItem = ({ icon, text, active, onClick }: ToolboxItemProps) => {
+  const style: string =
+    'flex cursor-pointer flex-col items-center justify-center gap-2 border-black p-1 text-center select-none hover:text-primary hover:fill-primary';
+
+  return (
+    <div className={classNames(style, active && ' border font-bold')} onClick={onClick}>
+      <div>{icon}</div>
+      <div className="text-xs">{text}</div>
+    </div>
+  );
+};
+
+const ToolboxColorPicker = ({ onColorChange, handleClose }: ToolboxColorPickerProps) => {
+  // to jako jeszcze jeden typ czy obiekt tutaj?? Wtedy jak ma być typ (ColorClass) z typu (Color)?
```

Comment:
tak samo jak kolory w consts/color, nie musisz tego wynosić aż do /src, może to być trzymane tutaj blisko komponentu. Czyli przenosisz ten obiekt do pliku consts, tworzysz na jego podstawie nowy typ i wtedy używasz tego typu do komponentów które akceptują taki parametr